typescript/web/src/components/pagination-tool/pagination-tool.tsx

```diff
@@ -0,0 +1,134 @@
+import {
+  // useColorModeValue as mode,
+  HStack,
+  IconButton,
+  NumberInput,
+  NumberInputField,
+  Tooltip,
+  Text,
+} from "@chakra-ui/react";
+import { RiArrowLeftSLine, RiArrowRightSLine } from "react-icons/ri";
+import { isNaN, clamp } from "lodash/fp";
+import { useEffect, useState } from "react";
+
+interface PaginationToolProps {
+  currentPage: number;
+  maxPage: number;
+  onChange: (newPage: number) => void;
+}
+
+const digitsPerRem = 0.55;
+
+export const PaginationTool = (props: PaginationToolProps) => {
+  const { currentPage, maxPage, onChange: setPage } = props;
+  const [inputValue, setInputValue] = useState(currentPage);
+  const digitCount = Math.max(1, Math.ceil(Math.log10(maxPage ?? 1)));
+
+  useEffect(() => {
+    setInputValue(currentPage);
+  }, [currentPage]);
+
+  const handleKeyPress = (event: any) => {
+    if (event.key === "Enter") {
+      setPage(event.target.value as number);
+    }
+    if (event.key === "Escape" || event.key === "Esc") {
+      event.preventDefault();
+      setInputValue(currentPage);
+    }
+  };
+
+  return (
+    <HStack h={10} p={0} spacing={1} rounded={6} pointerEvents="initial">
+      {currentPage > 1 ? (
+        <Tooltip
+          openDelay={300}
+          label={`Previous Page: ${currentPage - 1}`}
+          placement="top"
+        >
+          <IconButton
+            bg="none"
+            aria-label="Previous page"
+            icon={<RiArrowLeftSLine size="1.5em" />}
+            onClick={() => setPage(currentPage - 1)}
+          />
+        </Tooltip>
+      ) : (
+        <IconButton
+          disabled
+          bg="none"
+          aria-label="No previous page"
+          icon={<RiArrowLeftSLine size="1.5em" />}
+        />
+      )}
+      <Tooltip label="Current page" placement="top" openDelay={300}>
+        <NumberInput
+          name="current-page"
+          rounded={6}
+          allowMouseWheel
+          defaultValue={currentPage != null ? currentPage : 1}
+          min={1}
+          max={maxPage}
+          variant="filled"
+          textAlign="center"
+          size="sm"
+          value={inputValue}
+          onChange={(valStr, valNum) => {
+            if (!isNaN(valNum)) {
+              setInputValue(clamp(valNum, 1, maxPage));
+            }
+          }}
+          onKeyPress={handleKeyPress}
+          onBlur={() => {
+            setPage(inputValue);
+          }}
+        >
+          <NumberInputField
```

Comment:
We can't erase the number completely to set a new number, we have to set the new number and then delete the precedent one. Example: if I'm on page one and I want to type 2 to go to page 2, I need to first type number 2, so for a moment I have the number 12 in there, and then delete the 1